### PR TITLE
feat(sessions): add trace latency p50 to session details

### LINKED ISSUE #5232: [sessions][gql] add session trace latency to project sessions type

### PR DESCRIPTION
**Pull request by Parker-Stafford**
resolves #5232 

- adds trace_latency_ms_quantile field onto project session
- adds p50 to session details in ui

image

### CODE REVIEW

**Review by RogerHYang — APPROVED** [review]

File: src/phoenix/server/api/types/ProjectSession.py [hunk 1]
```diff
@@ -103,6 +110,23 @@ async def traces(
...
+        probability: float,
+    ) -> Optional[float]:
+        return await info.context.data_loaders.latency_ms_quantile.load(
+            (
+                "trace",
+                self.project_rowid,
+                None,
+                None,
+                probability,
+                models.Trace.project_session_rowid == self.project_rowid,
```
Comment (RogerHYang) [inline src/phoenix/server/api/types/ProjectSession.py:126]:
```suggestion
                models.Trace.project_session_rowid == self.id_attr,
```

Comment (Parker-Stafford) [reply src/phoenix/server/api/types/ProjectSession.py:126]: oh wow good catch

**Comment by RogerHYang** [comment]
LGTM. Nice work!

**Review by RogerHYang — APPROVED** [review]

File: src/phoenix/server/api/dataloaders/session_trace_latency_ms_quantile.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        session_rowids = {session_id for session_id, _ in keys}
+        for position, (session_id, probability) in enumerate(keys):
+            argument_position_map[session_id][probability].append(position)
+        stmt = (
+            select(
+                models.Trace.project_session_rowid,
+                models.Trace.latency_ms,
+            )
+            .where(models.Trace.project_session_rowid.in_(session_rowids))
+            .order_by(models.Trace.project_rowid)
```
Comment (RogerHYang) [inline src/phoenix/server/api/dataloaders/session_trace_latency_ms_quantile.py:43]:
```suggestion
            .order_by(models.Trace.project_session_rowid)
```

File: tests/unit/server/api/dataloaders/conftest.py [hunk 3]
```diff
@@ -14,80 +14,98 @@ async def data_for_testing_dataloaders(
...
-                end_time = orig_time + timedelta(seconds=seconds * K * 2)
-                trace_row_id = await session.scalar(
-                    insert(models.Trace)
+                end_time = orig_time + timedelta(seconds=seconds * l * 2)
+                session_row_id = await session.scalar(
+                    insert(models.ProjectSession)
                     .values(
-                        trace_id=f"{i}_{j}",
-                        project_rowid=project_row_id,
+                        session_id=f"{i}_{l}",
```
Comment (RogerHYang) [inline tests/unit/server/api/dataloaders/conftest.py:33]: nit. I was lazy, but you don't have to be. lol
```suggestion
                        session_id=f"proj{i}_sess{l}",
```

File: src/phoenix/server/app.py [hunk 4]
```diff
@@ -95,6 +95,9 @@
     UserRolesDataLoader,
     UsersDataLoader,
 )
+from phoenix.server.api.dataloaders.session_trace_latency_ms_quantile import (
```
Comment (RogerHYang) [inline src/phoenix/server/app.py:98]: nit
```suggestion
from phoenix.server.api.dataloaders import (
```

File: src/phoenix/server/api/context.py [hunk 5]
```diff
@@ -44,6 +44,9 @@
     UserRolesDataLoader,
     UsersDataLoader,
 )
+from phoenix.server.api.dataloaders.session_trace_latency_ms_quantile import (
```
Comment (RogerHYang) [inline src/phoenix/server/api/context.py:47]: nit
```suggestion
from phoenix.server.api.dataloaders import (
```

**Comment by Parker-Stafford** [comment]
> Can you also add a small test to [this file](https://github.com/Arize-ai/phoenix/blob/sessions2/tests/unit/server/api/types/test_ProjectSession.py)? A minimal one will suffice. Thanks.

done!

**Pull request merged**